Source/JavaScriptCore/bytecompiler/NodesCodegen.cpp

```diff
@@ -4982,6 +4989,12 @@ void FuncDeclNode::emitBytecode(BytecodeGenerator& generator, RegisterID*)
 
 RegisterID* FuncExprNode::emitBytecode(BytecodeGenerator& generator, RegisterID* dst)
 {
+    if (isStaticBlockFunction()) {
+        RefPtr<RegisterID> homeObject = emitHomeObjectForCallee(generator);
+        RefPtr<RegisterID> res = generator.emitNewFunctionExpression(generator.finalDestination(dst), this);
+        emitPutHomeObject(generator, res.get(), homeObject.get());
+        return res.get();
+    }
```

Comment:
I think, instead of having this flag here, we should do this in the caller side of this code generation.